### ISSUE TITLE
Consider moving the "Roles" checkboxes into a vertical tab

### ISSUE THREAD

**Issue by klonos** [backdrop-contrib/ckeditor5#16]
Now that the various settings for image/file uploads etc. have been moved into vtabs, perhaps we should also move the "Roles" checkbox set into a vtab as well. I think that this should be the first/default tab.

**Comment by quicksketch** [comment]
The "Roles" checkboxes are a text-format level configuration, not an editor-level configuration. Note that the Roles element is shown even when no editor is selected. Although editor-providing modules get passed the entire `$form` object, I think it'd be strange to move Roles into the vertical tabs of editor-specific settings.

**Comment by klonos** [comment]
> The "Roles" checkboxes are a text-format level configuration... it'd be strange to move Roles into the vertical tabs of editor-specific settings.

From an end-user point of view though, this is just another form element. I believe that only people that understand the technical distinction behind these things might find that odd - the rest would simply appreciate the consolidated form with less scrolling required to get to the filter settings. I think that this is another case where we are exposing things in the UI that are developer-related and shouldn't matter to the person that is using the software.

> Note that the Roles element is shown even when no editor is selected.

Yup, and we could be moving that outside any fieldset/vtab if no editor is selected. That would be similar to what we are doing with the user create/edit form, where with https://github.com/backdrop/backdrop-issues/issues/3872 we have placed the roles and the account status in an always-expanded fieldset (merely for grouping purposes):
![image](https://github.com/quicksketch/ckeditor5/assets/2423362/126b13c3-f9f5-43a2-b4a2-b4a34cc7b953)

...we however place everything in a vtab set once more settings become available in the form (after enabling locale/language modules etc), which reduces scrolling and provides a much better UX:
![image](https://github.com/quicksketch/ckeditor5/assets/2423362/3f3410bb-fe9f-4bc9-97eb-02eab38875c0)

**Comment by herbdool** [comment]
I don't think this should be a blocker from getting an official release, considering we'd like this module to be mostly function in a shortish timeline

**Comment by wimleers** [comment]
Sibling Drupal issue: https://www.drupal.org/project/drupal/issues/2569741
